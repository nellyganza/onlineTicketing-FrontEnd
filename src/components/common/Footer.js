import React from "react";

class fotter extends React.Component {
  render() {
    return (
      <footer className="section footer noover">
        <div className="container">
          <div className="row">
            <div className="col-lg-4 col-md-4">
              <div className="widget clearfix">
                <h3 className="widget-title">Subscribe Our Channel</h3>
                <div className="newsletter-widget">
                  <p>
                    You can opt out of our newsletters at any time.
                    <br /> See our <a>privacy policy</a>.
                  </p>
                </div>
                {}
              </div>
              {}
            </div>
            {}
            <div className="col-lg-3 col-md-3">
              <div className="widget clearfix">
                <h3 className="widget-title">Join us today</h3>
                <p>
                  Would you like to earn your profits by joining our team? Join
                  us without losing time.
                </p>
              </div>
              {}
            </div>
            {}
            <div className="col-lg-3 col-md-3">
              <div className="widget clearfix">
                <h3 className="widget-title">Popular Tags</h3>
                <div className="tags-widget">
                  <ul className="list-block">
                    <li>
                      <a>Live stream</a>
                    </li>
                    <li>
                      <a>Video Coverage</a>
                    </li>
                    <li>
                      <a>Event's Ticket</a>
                    </li>
                    <li>
                      <a>Pre-Record</a>
                    </li>
                  </ul>
                </div>
                {}
              </div>
              {}
            </div>
            {}
            <div className="col-lg-2 col-md-2">
              <div className="widget clearfix">
                <h3 className="widget-title">Support</h3>
                <div className="list-widget">
                  <ul>
                    <li>
                      <a>Terms of Use</a>
                    </li>
                    <li>
                      <a>Copyrights</a>
                    </li>
                    <li>
                      <a>Create a Ticket</a>
                    </li>
                    <li>
                      <a>Pricing & Plans</a>
                    </li>
                  </ul>
                </div>
                {}
              </div>
              {}
            </div>
            {}
          </div>
          {}
        </div>
        {}
      </footer>
    );
  }
}

export default fotter;
