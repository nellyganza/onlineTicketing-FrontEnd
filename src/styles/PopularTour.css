@import url('https://fonts.googleapis.com/css?family=Allura|Josefin+Sans');
.wrapper{
  margin-top: 50px;
}

#somePeople{
  font-size: 1.6rem;
  margin-bottom: 7rem;
  letter-spacing: normal;
  text-align: center;
}

.team{
  display: flex;
  justify-content: center;
  width: auto;
  text-align: center;
  flex-wrap: wrap;
}

.team .team_member{
  background: #fff;
  margin: 5px;
  margin-bottom: 50px;
  width: 300px;
  padding: 20px;
  line-height: 20px;
  color: #8e8b8b;  
  position: relative;
}

.team .team_member h3{
  color: #1420c7;
  font-size: 26px;
  margin-top: 50px;
}

.team .team_member p.role{
  color: #ccc;
  margin: 12px 0;
  font-size: 12px;
  text-transform: uppercase;
}

.team .team_member .team_img{
  position: absolute;
  top: -50px;
  left: 50%;
  transform: translateX(-50%);
  width: 100px;
  height: 100px;
  border-radius: 50%;
  background: #fff;
}

.team .team_member .team_img img{
  width: 100px;
  height: 100px;
  padding: 5px;
}

.flexBody{
    display: block;
    justify-content: center;
    align-items: center;
    height: auto;
    overflow: hidden;


}
.container{
    position: relative;
    width: 1100px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-wrap: wrap;
    padding: 30px;
}
.container .card{
    position: relative;
    max-width: 350px;
    height: 215px;
    margin: 30px 10px;
    padding: 20px 14px;
    display: flex;
    flex-direction: column;
    box-shadow: 0 5px 202px rgba(0, 0,0,0.5);
    transition: 0.3s ease-in-out;

}

.container .card:hover{
  height: 420px;

}

.container .card .imgBox{
position: relative;
width: 300px;
height: 250px;
top:-60px;
left: 20px;
z-index: 1;
box-shadow: rgba(0, 0,0,0.5);
}
.container .card .imgBox .img{
    max-width: 100%;
    border-radius: 50;
}
.container .card .content{
    position: relative;
    margin-top: -140px;
    padding: 5px 7px;
    text-align: center;
    color: #111;
    visibility: hidden;
    opacity: 0;
    transition: 0.3s ease-in-out;
}

.container .card:hover .content{
    visibility: visible;
    opacity: 1;
    margin-top: 15px;
    transition-delay: 0.3s;
    font-size: 1rem;
}
.content-h2{
    font-size: 16px;
    font-weight: bold;
}
.preMain
.preMain::before,
.preMain::after {
  box-sizing: border-box;
}

.main{
  max-width: 1200px;
  margin: 0 auto;
}

h1 {
    font-size: 24px;
    font-weight: 400;
    text-align: center;
}

img {
  height: auto;
  max-width: 100%;
  vertical-align: middle;
}

.btn {
  color: #ffffff;
  padding: 0.8rem;
  font-size: 14px;
  text-transform: uppercase;
  border-radius: 4px;
  font-weight: 400;
  display: block;
  width: 100%;
  cursor: pointer;
  border: 1px solid rgba(255, 255, 255, 0.2);
  background: transparent;
}

.btn:hover {
  background-color: rgba(20, 216, 13, 0.6);
}

.cards {
  display: flex;
  flex-wrap: wrap;
  list-style: none;
  margin: 0;
  padding: 0;
}

.cards_item {
  display: flex;
  padding: 1rem;
}

@media (min-width: 40rem) {
  .cards_item {
    width: 50%;
  }
}

@media (min-width: 56rem) {
  .cards_item {
    width: 33.3333%;
  }
 
}

@media (max-width: 56rem) {

  #hide{
    visibility: hidden;
  }
 
}

.card2 {
  background-color: white;
  border-radius: 0.25rem;
  box-shadow: 0 20px 40px -14px rgba(0, 0, 0, 0.25);
  display: flex;
  flex-direction: column;
  overflow: hidden;
}

.card_content {
  padding: 1rem;
  background: linear-gradient(to bottom left, #4240d4 40%, #4ca885 100%);
}

.card_title {
  color: #ffffff;
  font-size: 1.1rem;
  font-weight: 700;
  letter-spacing: 1px;
  text-transform: capitalize;
  margin: 0px;
}

.card_text {
  color: #ffffff;
  font-size: 0.875rem;
  line-height: 1.5;
  margin-bottom: 1.25rem;    
  font-weight: 400;
}
.made_by{
  font-weight: 400;
  font-size: 13px;
  margin-top: 35px;
  text-align: center;
}
hr.style18 { 
  height: 30px; 
  border-style: solid; 
  border-color: #8c8b8b; 
  border-width: 1px 0 0 0; 
  border-radius: 20px; 
} 
hr.style18:before { 
  display: block;
  content: ""; 
  height: 30px; 
  margin-top: -31px; 
  border-style: solid; 
  border-color: #8c8b8b; 
  border-width: 0 0 1px 0; 
  border-radius: 20px; 
}