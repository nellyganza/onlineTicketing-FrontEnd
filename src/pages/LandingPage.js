import React from 'react';
import ClientRecomended from '../components/client';
import Footer from '../components/common/Footer';
import Header from '../components/common/header';
import VideoSection from '../components/HomePageComponents/videosection';
import WelcomeService from '../components/HomePageComponents/welcomeServices';
import UpcomingEvents from '../components/HomePageComponents/upcomingEvents';
import PreveousEvents from '../components/HomePageComponents/preveousEvents';
const landingPage = () => (
  <><Header/>
  <div className="w-screen space-y-10 overflow-hidden">
    
    <VideoSection />
    <WelcomeService />
    <UpcomingEvents />
    <PreveousEvents />
    <ClientRecomended className="mx-10" />
    <Footer /> 

  </div>
  </>
);

export default landingPage;
