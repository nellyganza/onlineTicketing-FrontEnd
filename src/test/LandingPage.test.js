import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import Footer from '../components/common/Footer';
import Topnav from '../components/common/TopNav';
import About from '../pages/About';
import Help from '../pages/Help';
import App from '../app';
import LandingPage from '../pages/LandingPage';
import Testimonials from '../components/client';
import LandingSection from '../components/LandingSection';
import DestinationPageContent from '../components/Destination';
import PopularTour from '../components/PoularTour';
import store from '../store';
import { mount } from 'enzyme/build';

describe('It should mount the App Components', () => {
  it('Navbar should render', () => {
    shallow(<App />);
  });
  it('Navbar should render landing Page', () => {
    shallow(<LandingPage />);
  });
  it('Destination should render', () => {
    shallow(<DestinationPageContent />);
  });

  it('Landing section should render', () => {
    shallow(<Provider store={store}><Router><LandingSection /></Router></Provider>);
  });
  it('About section should render', () => {
    shallow(<About />);
  });
  it('Help page should render', () => {
    shallow(<Help />);
  });
  it('Popular tour should render', () => {
    shallow(<PopularTour />);
  });
  it('Testimonials should render', () => {
    shallow(<Testimonials />);
  });
});

describe('Testing Shared Shared Components', () => {
  it('footer should render', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Landing page content should be rendered', () => {
    const tree = renderer.create(<Provider store={store}><Router><Topnav /></Router></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  const toggleaRole = jest.fn();

  it('button click should toggle on mobile view ', () => {
    const component = mount(<Provider store={store}><Router><Topnav /></Router></Provider>);
    component
      .find('button#toggle_button')
      .simulate('click');
    expect(toggleaRole).toMatchSnapshot();
  });
});
