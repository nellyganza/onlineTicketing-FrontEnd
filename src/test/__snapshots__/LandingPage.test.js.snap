// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Shared Shared Components Landing page content should be rendered 1`] = `
<div
  className="absolute w-full h-auto z-10"
>
  <nav
    className="flex justify-between bg-white w-full md:w-topNav  p-3 md:py-3 md:pl-4  md:pr-28 h-full md:rounded-br-full"
  >
    <a
      className="block"
      href="/"
      onClick={[Function]}
    >
      <span
        className="text-2xl text-black font-bold  font-Sansation lg:pl-12 italic"
      >
        BareFoot
      </span>
    </a>
    <button
      className="lg:hidden block btn-submit"
      id="toggle_button"
      onClick={[Function]}
      type="button"
    >
      <svg
        height="24"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M3 15h18v-2H3v2zm0 4h18v-2H3v2zm0-8h18V9H3v2zm0-6v2h18V5H3z"
        />
      </svg>
    </button>
    <div
      className="hidden lg:flex  lg:items-center bg-white lg:w-auto w-full"
    >
      <section>
        <ul
          className="lg:flex items-center justify-between text-base text-black font-Sansation pt-4 lg:pt-0 "
        >
          <li>
             
            <a
              className="py-1 px-5 hover:bg-gray-200 rounded-full"
              href="/about"
              onClick={[Function]}
            >
              About
            </a>
          </li>
          <li>
             
            <a
              className="py-1 px-5 hover:bg-gray-200 rounded-full"
              href="/login"
              onClick={[Function]}
            >
              Accomodation
            </a>
          </li>
          <li>
             
            <a
              className="py-1 px-5 hover:bg-gray-200 rounded-full"
              href="/help"
              onClick={[Function]}
            >
              Help
            </a>
          </li>
          <li>
             
            <a
              className="inline-block px-6 py-2 text-sm font-bold leading-6 text-center text-white uppercase transition bg-green-400 rounded-full shadow ripple hover:shadow-lg hover:bg-green-500 focus:outline-none"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
      </section>
    </div>
  </nav>
  <div
    className="hidden"
  >
    <section
      className="text-center z-0 lg:items-center bg-white lg:w-auto"
    >
      <ul
        className="lg:hidden items-center justify-between text-base text-black font-bold pt-4 lg:pt-0 mb-4"
      >
        <li
          className="hover:bg-gray-200"
        >
           
          <a
            className="py-1 px-5"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
        <li
          className="hover:bg-gray-200"
        >
           
          <a
            className="py-1 px-5"
            href="/login"
            onClick={[Function]}
          >
            Accomodation
          </a>
        </li>
        <li
          className="hover:bg-gray-200"
        >
           
          <a
            className="py-1 px-5"
            href="/help"
            onClick={[Function]}
          >
            help
          </a>
        </li>
        <li
          className="hover:bg-gray-200"
        >
           
          <a
            className="py-1 px-5 bg-green-500 rounded-full"
            href="/login"
            onClick={[Function]}
          >
            Login
          </a>
        </li>
      </ul>
    </section>
  </div>
</div>
`;

exports[`Testing Shared Shared Components button click should toggle on mobile view  1`] = `[MockFunction]`;

exports[`Testing Shared Shared Components footer should render 1`] = `
<footer
  className="bg-footer-bg relative bg-no-repeat bg-cover lg:bg-center p-0 mb-0 bg-top-right"
>
  <div
    className="flex-1 max-w-8xl max-auto p-10"
  >
    <ul
      className="md:grid sm:grid-flow-col mt-30 grid-col-1 gap-4 md:grid-cols-2 md:gap-8 grid-flow-col-dense sm:auto-cols-min mb-2"
    >
      <li
        className="rounded-lg  flex-col md:p-12"
      >
        <div
          className="max-h-full"
        >
          <h1
            className="text-gray-200 font-bold text-4xl font-Sansation text-left justify-start"
          >
            BareFoot
          </h1>
          <p
            className="max-h-full p-1 pt-5 mb-2  text-gray-200 font-bold md:text-lg font-Sansation"
          >
            Barefoot is a grobal travel and accomodation website for nomads we make your travel experience more wonderful with our incomperable services.
          </p>
        </div>
      </li>
      <li
        className="rounded-lg  md:mr-12 md:p-12"
      >
        <div
          className="max-h-full"
        >
          <h1
            className="text-gray-200 font-bold text-4xl font-Sansation text-left justify-start"
          >
            Contact
          </h1>
          <p
            className="text-gray-200 pt-5 font-bold md:text-lg font-Sansation"
          >
            Barefoot customer services
            <br />
             
            +250785855328
            <br />
            barefoot@gmail.com
          </p>
          <div
            className="flex mt-4"
          >
            <a
              aria-label="link"
              href="https://twitter.com/BarefootCollege"
            >
              <svg
                className="text-white mr-4 text-2xl mb-5 hover:text-green-400"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                version="1.1"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 3.538c-0.588 0.263-1.222 0.438-1.884 0.516 0.678-0.406 1.197-1.050 1.444-1.816-0.634 0.375-1.338 0.65-2.084 0.797-0.6-0.638-1.453-1.034-2.397-1.034-1.813 0-3.281 1.469-3.281 3.281 0 0.256 0.028 0.506 0.084 0.747-2.728-0.138-5.147-1.444-6.766-3.431-0.281 0.484-0.444 1.050-0.444 1.65 0 1.138 0.578 2.144 1.459 2.731-0.538-0.016-1.044-0.166-1.488-0.409 0 0.013 0 0.028 0 0.041 0 1.591 1.131 2.919 2.634 3.219-0.275 0.075-0.566 0.116-0.866 0.116-0.212 0-0.416-0.022-0.619-0.059 0.419 1.303 1.631 2.253 3.066 2.281-1.125 0.881-2.538 1.406-4.078 1.406-0.266 0-0.525-0.016-0.784-0.047 1.456 0.934 3.181 1.475 5.034 1.475 6.037 0 9.341-5.003 9.341-9.341 0-0.144-0.003-0.284-0.009-0.425 0.641-0.459 1.197-1.038 1.637-1.697z"
                />
              </svg>
            </a>
            <a
              aria-label="link"
              href="https://web.facebook.com/groups/217860805077474"
            >
              <svg
                className="text-white mr-4 text-2xl mb-5 hover:text-green-500 "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                version="1.1"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.5 3h2.5v-3h-2.5c-1.93 0-3.5 1.57-3.5 3.5v1.5h-2v3h2v8h3v-8h2.5l0.5-3h-3v-1.5c0-0.271 0.229-0.5 0.5-0.5z"
                />
              </svg>
            </a>
            <a
              aria-label="link"
              href="https://www.instagram.com/barefoot_nomad_/"
            >
              <svg
                className="text-white mr-4 text-2xl mb-5 hover:text-green-500 "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                version="1.1"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14.5 0h-13c-0.825 0-1.5 0.675-1.5 1.5v13c0 0.825 0.675 1.5 1.5 1.5h13c0.825 0 1.5-0.675 1.5-1.5v-13c0-0.825-0.675-1.5-1.5-1.5zM11 2.5c0-0.275 0.225-0.5 0.5-0.5h2c0.275 0 0.5 0.225 0.5 0.5v2c0 0.275-0.225 0.5-0.5 0.5h-2c-0.275 0-0.5-0.225-0.5-0.5v-2zM8 5c1.656 0 3 1.344 3 3s-1.344 3-3 3c-1.656 0-3-1.344-3-3s1.344-3 3-3zM14 13.5v0c0 0.275-0.225 0.5-0.5 0.5h-11c-0.275 0-0.5-0.225-0.5-0.5v0-6.5h1.1c-0.066 0.322-0.1 0.656-0.1 1 0 2.762 2.237 5 5 5s5-2.238 5-5c0-0.344-0.034-0.678-0.1-1h1.1v6.5z"
                />
              </svg>
            </a>
          </div>
        </div>
      </li>
    </ul>
  </div>
</footer>
`;
